### PR TITLE
[quant][graphmode][fx] Support inplace option

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #43984 [quant][graphmode][fx][api] Call fuse in prepare
* **#43983 [quant][graphmode][fx] Support inplace option**
* #44042 [quant][graphmode][fx] Support quantize per channel in all cases

Summary:
Support inplace option in apis

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23459260](https://our.internmc.facebook.com/intern/diff/D23459260)

### CODE REVIEW

File: torch/quantization/quantize_fx.py [hunk 1]
```diff
@@ -91,22 +91,18 @@ def _quantize_fx(model, qconfig_dict, run_fn=None, run_args=None, inplace=False,
 
     if is_dynamic_quant:
         model = prepare_dynamic_fx(model, qconfig_dict, inplace)
-        # TODO: change inplace to True since the model is already copied in
-        # prepare
-        model = convert_dynamic_fx(model, False, debug)
+        model = convert_dynamic_fx(model, inplace=True, debug=debug)
```
Comment (vkuzo) [inline torch/quantization/quantize_fx.py:94]: should this be `inplace=inplace`?

File: torch/quantization/quantize_fx.py [hunk 2]
```diff
@@ -91,22 +91,18 @@ def _quantize_fx(model, qconfig_dict, run_fn=None, run_args=None, inplace=False,
...
     else:
         assert run_fn, "Must provide calibration function for post training static quantization"
         assert run_args, "Must provide calibration dataset for post training static quantization"
         model = prepare_fx(model, qconfig_dict, inplace)
         run_fn(model, *run_args)
-        # TODO: change inplace to True since the model is already copied in
-        # prepare
-        model = convert_fx(model, False, debug)
+        model = convert_fx(model, inplace=True, debug=debug)
```
Comment (vkuzo) [inline torch/quantization/quantize_fx.py:100]: and here

File: torch/quantization/fx/quantize.py [hunk 3]
```diff
@@ -282,26 +281,24 @@ def _run_weight_observers(self, observed):
...
-        observed.eval().cpu()
-        observed_root = observed.root
-        observed_graph = observed.graph
-        if not inplace:
-            observed_root = copy.deepcopy(observed_root)
-
-        self.modules = dict(observed_root.named_modules())
+        model.eval().cpu()
+        model_root = model.root
+        model_graph = model.graph
```
Comment (vkuzo) [inline torch/quantization/fx/quantize.py:297]: optional nit: can we just use `model.graph`?

File: torch/quantization/fx/quantize.py [hunk 4]
```diff
@@ -514,10 +512,11 @@ def load_arg(a):
                 # copy other nodes
                 env[node.name] = folded_graph.node_copy(node, load_arg)
         folded_graph.output(load_arg(quantized_graph.result))
-        return GraphModule(quantized_root, folded_graph)
+        quantized.__init__(quantized_root, folded_graph)
+        return quantized
 
-    def convert(self, observed, inplace=False, debug=False, is_dynamic=False):
-        quantized = self._convert(observed, inplace, debug, is_dynamic)
+    def convert(self, model, inplace=False, debug=False, is_dynamic=False):
```
Comment (vkuzo) [inline torch/quantization/fx/quantize.py:518]: thanks for renaming, the new name is much clearer

**Review by vkuzo — APPROVED** [review]
lg, seems like the `inplace=inplace` should be fixed before landing

File: torch/quantization/quantize_fx.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/quantize_fx.py:94]: this should be True since the inplace option is already applied in the previous step(prepare)

**Pull request closed without merging**